### ISSUE TITLE
feature request: control module

### ISSUE THREAD

**Issue by enderman-2009** [cadergator10/Opencomputers-servertine#3]
could be multiple modules if you want
potential features
- door editing (maybe similar to diagnostic tablet
- door control (open/close control )
- sector control (lockdown/open control)

**Comment by enderman-2009** [comment]
@cadergator10 maybe also a color palette editor but idk if that is possible with the current version

**Comment by cadergator10** [comment]
I don't understand what you mean by this

**Comment by enderman-2009** [comment]
by what

**Comment by enderman-2009** [comment]
@cadergator10 what do you not understand

**Comment by cadergator10** [comment]
What does this control module do? Like is it just all 3 of the modules together?

**Comment by enderman-2009** [comment]
@cadergator10 some of this may require extra modification but also you can separate them if you think its better but its essentially this:
1. door editing: like diagnostic tablet door editing and the door setup module in one thing
2. door control: similar to remote control but maybe you could like store the addresses of the door computers or something so you don't have to walk to a mag reader everytime
3. like the sector control but from the database with buttons instead

**Comment by enderman-2009** [comment]
@cadergator10 i think this is a good idea cause editing doors with the diagnostic tablet is pain, the others the door control isn't require but sector control would be nice

**Comment by cadergator10** [comment]
OOOOOH :P I mistook what you said about the control module and assumed you meant the three modules already in place. 

Door Editing: I was planning on adding this feature to the current door editing, so you can not only create doors, but edit doors already in place.

Door control: Remote control could be something added, something almost exactly like the Diagnostic tablet. The diagnostic tablet's remote control doesn't require you to go to magreaders as well I dont think.

An ability to use the MineOS to control sectors is possible, but there are a few major caveats:
1.  The MineOS database is designed to be turned completely on and completely off, which makes redstone control pretty much impossible
2. Modules aren't designed much for background tasks. Really at most I could make it so you can control sectors manually, which leads into
3. The database sector control would override redstone sector control, and also right back at it. It's a whole mess of a problem.
Of course, workarounds can be found, but for the moment it doesn't make too much sense.

**Comment by enderman-2009** [comment]
btw i didn't mean redstone i meant like buttons or something

**Comment by enderman-2009** [comment]
and for the sector they still get overwriten if you use a bypass anyway

**Comment by cadergator10** [comment]
> and for the sector they still get overwriten if you use a bypass anyway

Yeah. A bypass I can think of is, rather than wait for when redstone is enabled, make it so when it detects a pulse it triggers a sector and add another redstone line in to disable lockdown or lock open.

**Comment by enderman-2009** [comment]
no a bypass like in the sectors module

**Comment by enderman-2009** [comment]
and i mean if someone doesn't want there sector control program to get overwriten then just don't use that feature

**Comment by cadergator10** [comment]
> no a bypass like in the sectors module

Sorry, I meant bypass like a way around this issue. Doing it that way would mean that sector statuses, instead of local to one sectorcontrol module, only update the server one (meaning multiple sectorcontrol devices are possible)

**Comment by cadergator10** [comment]
> also i think door editing should be the first of those features if you don't know where to start

It'll be a while until I add in any enhancements. My current progress is going towards the servertine website, so that when I begin work on the minecraft system, I can program it in to work with it

**Comment by enderman-2009** [comment]
@cadergator10 what is the servertine website?

**Comment by cadergator10** [comment]
It's the website I'm making for it. On launch it should allow you to

1. Add modules
2. View modules
3. Report modules

Basically will be where the data will be stored for the modules to be
downloaded (similar to MineOS' app store, but the interface is on a website
instead of the MineOS program in minecraft
It also allows creators to add their own modules easily without asking me
to add their modules in.

Over time, I'll add in more stuff like
1. A wiki (maybe)
2. Forum (like the OC forum)
3. If I'm feeling like it, some extra fun stuff with websockets and
accessing Servertine on minecraft through the website (EXTREME unlikely,
but if I want to have fun)

Currently, I have accounts working and avatars (as well as an ability for
me to review and reject/accept them if they violate guidelines)

In short: module downloading system similar to MineOS' app store

I do plan on making videos on my progress, but we'll see when I have time

**Comment by enderman-2009** [comment]
btw if i knew how to make modules i might make some

**Comment by cadergator10** [comment]
> btw if i knew how to make modules i might make some

I have created some documentation on how to make some (some changes to it will be made, but nothing that should break previous modules) and I'm always here to help. If you need my discord it's here cadergator10#6531

**Comment by enderman-2009** [comment]
i don't understand for me it just looks like a big mess of an explannation, also why would i need your discord?

**Comment by enderman-2009** [comment]
like i don't even know how to import custom modules in the system

**Comment by enderman-2009** [comment]
i have a decent amount of experience with lua i just don't understand how to make something for your system
like i made a network protocol that allows confirmation and a system to allow the transfer of files through that protocol extremely fast and also a remote code execution that works on both drone and a computer that also allows receiving messages from that location and it also has error protection it only crashes if it receives nil which in my setup only happens if you crash the program that sends the info

**Comment by cadergator10** [comment]
> i don't understand for me it just looks like a big incomprehensible mess, also why would i need your discord?

The discord is for instant communication if you want to ask me questions to be answered quicker, just for smaller stuff like help with the documentation (which I'll try and make a little better)

> like i don't even know how to import custom modules in the system

Once the new servertine module and server are released, it'll be much easier to import custom modules. Before, I had to manually wget the files to the server in order to debug it, but the new version will have a developer mode. The developer mode will use the second url specified by developers to download files (meaning download from the development branch of code) as well as backing up and clearing previous files to create a cleanish install of sorts.

> i have a decent amount of experience with lua i just don't understand how to make something for your system like i made a network protocol that allows confirmation and a system to allow the transfer of files through that protocol extremely fast and also a remote code execution that works on both drone and a computer that also allows receiving messages from that location and it also has error protection it only crashes if it receives nil which in my setup only happens if you crash the program that sends the info and more thats just the main things

Understandable lol. I need to put comments through my system and improve my documentation, which I plan on doing before the next release. Just a double check, have you seen the module prefabs for both the server and the database? If not, those could make it a little more easier. Also for the database, you need to understand the MineOS' GUI API.
